tests/protocols/test_http.py

```diff
@@ -527,6 +527,28 @@ def test_root_path(protocol_cls):
     assert b"Path: /app/" in protocol.transport.buffer
 
 
+@pytest.mark.parametrize("protocol_cls", HTTP_PROTOCOLS)
+def test_raw_path(protocol_cls):
+    async def app(scope, receive, send):
+        path = scope["path"]
+        raw_path = scope.get("raw_path", None)
+        response = Response(
+            "Path: {}\nraw_path: {}".format(path, repr(raw_path)),
+            media_type="text/plain",
+        )
+        await response(scope, receive, send)
+
+    protocol = get_connected_protocol(app, protocol_cls, root_path="/app")
+    protocol.data_received(
+        b"\r\n".join([b"GET /one%2Ftwo HTTP/1.1", b"Host: example.org", b"", b""])
```

Comment:
Let's have a `GET_REQUEST_WITH_RAW_PATH` constant or something similar, just to keep the style consistent with the surrounding test cases.

Reply:
Shall do.

Reply:
Done: https://github.com/encode/uvicorn/pull/372/commits/1c88abc91091a24fdc427ea8a9e973684f4c8e63

Reply:
Fixed: https://github.com/encode/uvicorn/pull/372/commits/6893373768a4e296e920ef44c9a4da93749a064f